bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RoundRobinDistributionSchedule.java

```diff
@@ -136,6 +221,7 @@ public QuorumCoverageSet getCoverageSet() {
 
     @Override
     public boolean hasEntry(long entryId, int bookieIndex) {
-        return getWriteSet(entryId).contains(bookieIndex);
+        int[] set = writeSets[(int) (entryId % ensembleSize)];
```

Comment:
I think this can be further improved. because it is just checking whether an entry is supposed to be stored in one bookie. I think we can simply calculate it.